Visual Studio Code (1.79.2, undefined, desktop)
Jupyter Extension Version: 2023.4.1001091014.
Python Extension Version: 2023.8.0.
Platform: linux (x64).
Workspace folder ~/study/python/lab, Home = /home/_wajs_err_
23:48:58.550 [info] ZMQ loaded without the fallback mechanism.
23:48:58.711 [debug] Get Custom Env Variables, Class name = wc (started execution), Arg 1: undefined, Arg 2: "RunPythonCode"
23:48:58.712 [info] Start refreshing Kernel Picker (1691009338712)
23:48:58.716 [info] Using Pylance
23:48:58.863 [debug] No controller, hence notebook communications cannot be initialized for editor ~/study/python/lab/lab.ipynb
23:48:59.085 [debug] Get Custom Env Variables, Class name = wc, completed in 374ms, has a truthy return value, Arg 1: undefined, Arg 2: "RunPythonCode"
23:48:59.086 [debug] Jupyter Paths /kernels: 
23:48:59.087 [debug] Kernel Spec Root Paths, /usr/share/jupyter/kernels, /usr/local/share/jupyter/kernels, ~/.local/share/jupyter/kernels
23:48:59.104 [debug] Getting activated env variables, Class name = Is (started execution), Arg 1: "~/study/python/lab", Arg 2: <Uri:/bin/python>, Arg 3: undefined
23:48:59.104 [debug] Get Custom Env Variables, Class name = wc (started execution), Arg 1: "~/study/python/lab", Arg 2: "RunPythonCode"
23:48:59.106 [debug] Get Custom Env Variables, Class name = wc, completed in 1ms, has a truthy return value, Arg 1: "~/study/python/lab", Arg 2: "RunPythonCode"
23:48:59.128 [debug] Not Searching for kernels as path does not exist, /usr/share/jupyter/kernels
23:48:59.128 [debug] Not Searching for kernels as path does not exist, /usr/local/share/jupyter/kernels
23:48:59.161 [debug] Loading kernelspec from ~/.local/share/jupyter/kernels/python3/kernel.json for 
23:48:59.162 [debug] Conda file returned by Python Extension is conda
23:48:59.293 [warn] Failed to get activated env vars for /bin/python in 189ms
23:48:59.295 [debug] Create ProcessService, Class name = nc (started execution), Arg 1: undefined
23:48:59.295 [debug] Get Custom Env Variables, Class name = wc (started execution), Arg 1: undefined, Arg 2: "RunNonPythonCode", Arg 3: undefined
23:48:59.298 [debug] Get Custom Env Variables, Class name = wc, completed in 3ms, has a truthy return value, Arg 1: undefined, Arg 2: "RunNonPythonCode", Arg 3: undefined
23:48:59.298 [debug] Create ProcessService, Class name = nc, completed in 3ms, has a truthy return value, Arg 1: undefined
23:48:59.314 [info] Process Execution: /bin/python -c "import site;print("USER_BASE_VALUE");print(site.USER_BASE);print("USER_BASE_VALUE");"
23:48:59.335 [debug] USER_SITE for /bin/python is ~/.local/bin
23:48:59.335 [debug] Prepend PATH with user site path for /bin/python, user site ~/.local/bin
23:48:59.335 [debug] Prepend PATH with python bin for /bin/python, 
    PATH value is ~/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl and 
    Path value is undefined
23:48:59.335 [debug] Activated Env Variables for /bin/python, 
    PATH value is /bin:~/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl and 
    Path value is undefined
23:48:59.335 [debug] Got env vars with python /bin/python in 231ms with 48 variables
23:48:59.335 [debug] Getting activated env variables, Class name = Is, completed in 231ms, has a truthy return value, Arg 1: "~/study/python/lab", Arg 2: <Uri:/bin/python>, Arg 3: undefined
23:48:59.346 [info] Process Execution: /bin/python -m pip list
23:48:59.457 [debug] Full interpreter list after refreshing (total 394ms, resolve 20ms, refresh 374ms) is length: 4, /bin/python:undefined::/bin/python:/bin/python, /sbin/python:undefined::/sbin/python:/sbin/python, /usr/bin/python:undefined::/usr/bin/python:/usr/bin/python, /usr/sbin/python:undefined::/usr/sbin/python:/usr/sbin/python
23:48:59.462 [debug] Full interpreter list after refreshing (total 1ms, resolve 1ms, refresh 0ms) is length: 4, /bin/python:undefined::/bin/python:/bin/python, /sbin/python:undefined::/sbin/python:/sbin/python, /usr/bin/python:undefined::/usr/bin/python:/usr/bin/python, /usr/sbin/python:undefined::/usr/sbin/python:/usr/sbin/python
23:48:59.466 [debug] refreshData after detecting changes to interpreters
23:48:59.467 [debug] Listing Python & non-Python kernels for Interpreter /bin/python
23:48:59.467 [debug] Listing Python & non-Python kernels for Interpreter /sbin/python
23:48:59.468 [debug] Finding Global Python KernelSpecs
23:48:59.470 [debug] refreshData after detecting changes to interpreters
23:48:59.470 [debug] Listing Python & non-Python kernels for Interpreter /usr/bin/python
23:48:59.471 [debug] refreshData after detecting changes to interpreters
23:48:59.471 [debug] Listing Python & non-Python kernels for Interpreter /usr/sbin/python
23:48:59.471 [debug] refreshData after detecting changes to interpreters
23:48:59.473 [debug] refreshData after detecting changes to interpreters
23:48:59.474 [debug] Full interpreter list is length: 4, /bin/python::Unknown:/bin/python, /sbin/python::Unknown:/sbin/python, /usr/bin/python::Unknown:/usr/bin/python, /usr/sbin/python::Unknown:/usr/sbin/python
23:48:59.474 [debug] Refreshed Environments
23:48:59.479 [debug] Kernel for interpreter /bin/python is .jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher
23:48:59.480 [debug] Kernels for interpreter /bin/python are .jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher
23:48:59.483 [debug] Kernel for interpreter /sbin/python is .jvsc74a57bd0d2152fd7f0bbc62aa1baff8c990435d1e2c7175d001561303988032604c11a48./sbin/python./sbin/python.-m#ipykernel_launcher
23:48:59.483 [debug] Kernels for interpreter /sbin/python are .jvsc74a57bd0d2152fd7f0bbc62aa1baff8c990435d1e2c7175d001561303988032604c11a48./sbin/python./sbin/python.-m#ipykernel_launcher
23:48:59.484 [debug] Kernel for interpreter /usr/bin/python is .jvsc74a57bd0767d51c1340bd893661ea55ea3124f6de3c7a262a8b4abca0554b478b1e2ff90./usr/bin/python./usr/bin/python.-m#ipykernel_launcher
23:48:59.484 [debug] Kernels for interpreter /usr/bin/python are .jvsc74a57bd0767d51c1340bd893661ea55ea3124f6de3c7a262a8b4abca0554b478b1e2ff90./usr/bin/python./usr/bin/python.-m#ipykernel_launcher
23:48:59.486 [debug] Kernel for interpreter /usr/sbin/python is .jvsc74a57bd04e1d9a8909477db77738c33245c29c7265277ef753467dede8cf3f814cde494e./usr/sbin/python./usr/sbin/python.-m#ipykernel_launcher
23:48:59.486 [debug] Kernels for interpreter /usr/sbin/python are .jvsc74a57bd04e1d9a8909477db77738c33245c29c7265277ef753467dede8cf3f814cde494e./usr/sbin/python./usr/sbin/python.-m#ipykernel_launcher
23:48:59.491 [debug] KernelProvider switched kernel to id = .jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher
23:48:59.492 [debug] start the kernel, options.disableUI=true for ~/study/python/lab/lab.ipynb
23:48:59.493 [debug] Initialize notebook communications for editor ~/study/python/lab/lab.ipynb
23:48:59.493 [debug] Resolving notebook UI Comms (resolve) for ~/study/python/lab/lab.ipynb
23:48:59.493 [debug] initialize CommonMessageCoordinator
23:48:59.498 [debug] Registering commtarget jupyter.widget
23:48:59.499 [debug] IPyWidgetMessageDispatcher.initialize
23:48:59.499 [debug] Attempting to determine version of IPyWidgets
23:48:59.500 [debug] Controller selection change completed
23:48:59.500 [debug] Waiting for IPyWidgets version
23:48:59.500 [debug] Waiting for IPyWidgets version promise
23:48:59.509 [info] Starting Kernel startUsingPythonInterpreter, .jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher  (Python Path: /bin/python, Unknown, 3.11.3) for '~/study/python/lab/lab.ipynb' (disableUI=true)
23:48:59.509 [debug] Creating raw notebook for resource '~/study/python/lab/lab.ipynb'
23:48:59.548 [debug] Kernel Launcher. launch, Class name = Kg (started execution), Arg 1: ".jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher", Arg 2: 60000, Arg 3: "~/study/python/lab/lab.ipynb", Arg 4: "~/study/python/lab", Arg 5: ""
23:48:59.548 [debug] Getting activated env variables, Class name = Is (started execution), Arg 1: "~/study/python/lab/lab.ipynb", Arg 2: <Uri:/bin/python>, Arg 3: undefined
23:48:59.549 [debug] Get Custom Env Variables, Class name = wc (started execution), Arg 1: "~/study/python/lab/lab.ipynb", Arg 2: "RunPythonCode"
23:48:59.557 [debug] Get Custom Env Variables, Class name = wc, completed in 8ms, has a truthy return value, Arg 1: "~/study/python/lab/lab.ipynb", Arg 2: "RunPythonCode"
23:48:59.574 [debug] Launching kernel in kernelProcess.ts, Class name = Ug (started execution), Arg 1: "~/study/python/lab", Arg 2: ""
23:48:59.576 [warn] Failed to get activated env vars for /bin/python in 27ms
23:48:59.577 [debug] Prepend PATH with user site path for /bin/python, user site ~/.local/bin
23:48:59.577 [debug] Prepend PATH with python bin for /bin/python, 
    PATH value is ~/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl and 
    Path value is undefined
23:48:59.577 [debug] Activated Env Variables for /bin/python, 
    PATH value is /bin:~/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl and 
    Path value is undefined
23:48:59.577 [debug] Got env vars with python /bin/python in 28ms with 48 variables
23:48:59.577 [debug] Getting activated env variables, Class name = Is, completed in 29ms, has a truthy return value, Arg 1: "~/study/python/lab/lab.ipynb", Arg 2: <Uri:/bin/python>, Arg 3: undefined
23:48:59.593 [info] Process Execution: /bin/python -c "import ipykernel; print(ipykernel.__version__); print("5dc3a68c-e34e-4080-9c3e-2a532b2ccb4d"); print(ipykernel.__file__)"
23:48:59.594 [debug] Kernel Env Variables for ~/.vscode-oss/extensions/ms-toolsai.jupyter-2023.4.1001091014-universal/temp/jupyter/kernels/pythonjvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a/kernel.json, PATH value is /bin:~/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
23:48:59.606 [info] Process Execution: /bin/python -m ipykernel_launcher --ip=127.0.0.1 --stdin=9003 --control=9001 --hb=9000 --Session.signature_scheme="hmac-sha256" --Session.key=b"18e152e9-f72a-4ecf-a838-ce898ab27f93" --shell=9002 --transport="tcp" --iopub=9004 --f=~/.local/share/jupyter/runtime/kernel-v2-23531003VLQySC5liU.json
    > cwd: ~/study/python/lab
23:48:59.606 [debug] Launching kernel in kernelProcess.ts, Class name = Ug, completed in 32ms, has a truthy return value, Arg 1: "~/study/python/lab", Arg 2: ""
23:48:59.622 [warn] Stderr output when getting ipykernel version & path Traceback (most recent call last):
    File "<string>", line 1, in <module>
    ModuleNotFoundError: No module named 'ipykernel' for /bin/python
23:48:59.630 [debug] KernelProcess error: /bin/python: No module named ipykernel_launcher

23:48:59.630 [warn] StdErr from Kernel Process /bin/python: No module named ipykernel_launcher
23:48:59.636 [debug] KernelProcess Exit Exit - 1 /bin/python: No module named ipykernel_launcher

23:48:59.637 [error] Disposing kernel process due to an error [mp [Error]: The kernel died. Error: /bin/python: No module named ipykernel_launcher... View Jupyter [log](command:jupyter.viewOutput) for further details.
	at ChildProcess.<anonymous> (~/.vscode-oss/extensions/ms-toolsai.jupyter-2023.4.1001091014-universal/out/extension.node.js:24:86433)
	at ChildProcess.emit (node:events:525:35)
	at ChildProcess._handle.onexit (node:internal/child_process:291:12)] {
  category: 'kerneldied',
  isJupyterError: true,
  kernelConnectionMetadata: Ku {
    kind: 'startUsingPythonInterpreter',
    kernelSpec: lu {
      specFile: '~/.vscode-oss/extensions/ms-toolsai.jupyter-2023.4.1001091014-universal/temp/jupyter/kernels/pythonjvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a/kernel.json',
      interpreterPath: '/bin/python',
      isRegisteredByVSC: 'registeredByNewVersionOfExt',
      name: 'pythonjvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a',
      argv: [Array],
      language: 'python',
      executable: 'python',
      display_name: 'Python 3',
      metadata: [Object],
      env: {},
      interrupt_mode: undefined
    },
    interpreter: {
      id: '/bin/python',
      sysPrefix: '/usr',
      envPath: undefined,
      displayPath: [n],
      envName: '',
      uri: [n],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher'
  },
  exitCode: 1,
  stdErr: '/bin/python: No module named ipykernel_launcher\n'
}
23:48:59.639 [info] Dispose Kernel process 235447.
23:48:59.639 [debug] Disposed Kernel process 235447.
23:48:59.640 [error] Kernel Launcher. launch, Class name = Kg, completed in 92ms, has a falsy return value, Arg 1: ".jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher", Arg 2: 60000, Arg 3: "~/study/python/lab/lab.ipynb", Arg 4: "~/study/python/lab", Arg 5: "" [mp [Error]: The kernel died. Error: /bin/python: No module named ipykernel_launcher... View Jupyter [log](command:jupyter.viewOutput) for further details.
	at ChildProcess.<anonymous> (~/.vscode-oss/extensions/ms-toolsai.jupyter-2023.4.1001091014-universal/out/extension.node.js:24:86433)
	at ChildProcess.emit (node:events:525:35)
	at ChildProcess._handle.onexit (node:internal/child_process:291:12)] {
  category: 'kerneldied',
  isJupyterError: true,
  kernelConnectionMetadata: Ku {
    kind: 'startUsingPythonInterpreter',
    kernelSpec: lu {
      specFile: '~/.vscode-oss/extensions/ms-toolsai.jupyter-2023.4.1001091014-universal/temp/jupyter/kernels/pythonjvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a/kernel.json',
      interpreterPath: '/bin/python',
      isRegisteredByVSC: 'registeredByNewVersionOfExt',
      name: 'pythonjvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a',
      argv: [Array],
      language: 'python',
      executable: 'python',
      display_name: 'Python 3',
      metadata: [Object],
      env: {},
      interrupt_mode: undefined
    },
    interpreter: {
      id: '/bin/python',
      sysPrefix: '/usr',
      envPath: undefined,
      displayPath: [n],
      envName: '',
      uri: [n],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher'
  },
  exitCode: 1,
  stdErr: '/bin/python: No module named ipykernel_launcher\n'
}
23:48:59.640 [error] Failed to connect raw kernel session: Error: The kernel died. Error: /bin/python: No module named ipykernel_launcher... View Jupyter [log](command:jupyter.viewOutput) for further details.
23:48:59.640 [debug] Shutdown session -- complete
23:48:59.640 [debug] failed to create IJupyterKernelConnectionSession in kernel, UI Disabled = true [mp [Error]: The kernel died. Error: /bin/python: No module named ipykernel_launcher... View Jupyter [log](command:jupyter.viewOutput) for further details.
	at ChildProcess.<anonymous> (~/.vscode-oss/extensions/ms-toolsai.jupyter-2023.4.1001091014-universal/out/extension.node.js:24:86433)
	at ChildProcess.emit (node:events:525:35)
	at ChildProcess._handle.onexit (node:internal/child_process:291:12)] {
  category: 'kerneldied',
  isJupyterError: true,
  kernelConnectionMetadata: Ku {
    kind: 'startUsingPythonInterpreter',
    kernelSpec: lu {
      specFile: '~/.vscode-oss/extensions/ms-toolsai.jupyter-2023.4.1001091014-universal/temp/jupyter/kernels/pythonjvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a/kernel.json',
      interpreterPath: '/bin/python',
      isRegisteredByVSC: 'registeredByNewVersionOfExt',
      name: 'pythonjvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a',
      argv: [Array],
      language: 'python',
      executable: 'python',
      display_name: 'Python 3',
      metadata: [Object],
      env: {},
      interrupt_mode: undefined
    },
    interpreter: {
      id: '/bin/python',
      sysPrefix: '/usr',
      envPath: undefined,
      displayPath: [n],
      envName: '',
      uri: [n],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher'
  },
  exitCode: 1,
  stdErr: '/bin/python: No module named ipykernel_launcher\n'
}
23:48:59.641 [warn] Error occurred while trying to start the kernel, options.disableUI=true [mp [Error]: The kernel died. Error: /bin/python: No module named ipykernel_launcher... View Jupyter [log](command:jupyter.viewOutput) for further details.
	at ChildProcess.<anonymous> (~/.vscode-oss/extensions/ms-toolsai.jupyter-2023.4.1001091014-universal/out/extension.node.js:24:86433)
	at ChildProcess.emit (node:events:525:35)
	at ChildProcess._handle.onexit (node:internal/child_process:291:12)] {
  category: 'kerneldied',
  isJupyterError: true,
  kernelConnectionMetadata: Ku {
    kind: 'startUsingPythonInterpreter',
    kernelSpec: lu {
      specFile: '~/.vscode-oss/extensions/ms-toolsai.jupyter-2023.4.1001091014-universal/temp/jupyter/kernels/pythonjvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a/kernel.json',
      interpreterPath: '/bin/python',
      isRegisteredByVSC: 'registeredByNewVersionOfExt',
      name: 'pythonjvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a',
      argv: [Array],
      language: 'python',
      executable: 'python',
      display_name: 'Python 3',
      metadata: [Object],
      env: {},
      interrupt_mode: undefined
    },
    interpreter: {
      id: '/bin/python',
      sysPrefix: '/usr',
      envPath: undefined,
      displayPath: [n],
      envName: '',
      uri: [n],
      displayName: '',
      envType: 'Unknown',
      isCondaEnvWithoutPython: false,
      version: [Object]
    },
    id: '.jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher'
  },
  exitCode: 1,
  stdErr: '/bin/python: No module named ipykernel_launcher\n'
}
23:48:59.940 [debug] Attempting to determine version of IPyWidgets
23:48:59.940 [debug] Waiting for IPyWidgets version
23:48:59.940 [debug] Waiting for IPyWidgets version promise
23:49:00.475 [info] End refreshing Kernel Picker (1691009338712)
23:49:14.488 [debug] Updating cache with Python kernels startUsingPythonInterpreter:'.jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher (interpreter id = /bin/python)', startUsingPythonInterpreter:'.jvsc74a57bd0d2152fd7f0bbc62aa1baff8c990435d1e2c7175d001561303988032604c11a48./sbin/python./sbin/python.-m#ipykernel_launcher (interpreter id = /sbin/python)', startUsingPythonInterpreter:'.jvsc74a57bd0767d51c1340bd893661ea55ea3124f6de3c7a262a8b4abca0554b478b1e2ff90./usr/bin/python./usr/bin/python.-m#ipykernel_launcher (interpreter id = /usr/bin/python)', startUsingPythonInterpreter:'.jvsc74a57bd04e1d9a8909477db77738c33245c29c7265277ef753467dede8cf3f814cde494e./usr/sbin/python./usr/sbin/python.-m#ipykernel_launcher (interpreter id = /usr/sbin/python)', Added = startUsingPythonInterpreter:'.jvsc74a57bd04e1d9a8909477db77738c33245c29c7265277ef753467dede8cf3f814cde494e./usr/sbin/python./usr/sbin/python.-m#ipykernel_launcher (interpreter id = /usr/sbin/python)', Updated = , Removed = 
23:49:18.515 [debug] notebook communications already initialized for editor ~/study/python/lab/lab.ipynb
23:49:46.244 [debug] notebook communications already initialized for editor ~/study/python/lab/lab.ipynb
23:49:51.336 [debug] notebook communications already initialized for editor ~/study/python/lab/lab.ipynb
23:49:55.901 [debug] notebook communications already initialized for editor ~/study/python/lab/lab.ipynb
23:49:58.842 [info] Shutting down notebooks for b980c8b3-aa6f-4592-89dd-0c7c3fd1d8e3
23:49:58.842 [info] Disposing HostJupyterExecution 4ff726d2-21df-4b04-bdd7-2e8fa1903e1e
23:49:58.842 [debug] Disposing super HostJupyterExecution 4ff726d2-21df-4b04-bdd7-2e8fa1903e1e
23:49:58.843 [info] Dispose Kernel '~/study/python/lab/lab.ipynb' associated with '~/study/python/lab/lab.ipynb'
23:49:58.843 [debug] Kernel got disposed, hence there is no longer a kernel associated with ~/study/python/lab/lab.ipynb
23:49:58.843 [info] Disposing HostJupyterExecution 4ff726d2-21df-4b04-bdd7-2e8fa1903e1e
23:49:58.843 [debug] Finished disposing HostJupyterExecution  4ff726d2-21df-4b04-bdd7-2e8fa1903e1e
23:49:58.844 [debug] Disposing controller .jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher associated with connection .jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher and documents file://~/study/python/lab/lab.ipynb
23:49:58.846 [debug] Disposing controller .jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher (Interactive) associated with connection .jvsc74a57bd0e7370f93d1d0cde622a1f8e1c04877d8463912d04d973331ad4851f04de6915a./bin/python./bin/python.-m#ipykernel_launcher 
23:49:58.846 [debug] Disposing controller .jvsc74a57bd0d2152fd7f0bbc62aa1baff8c990435d1e2c7175d001561303988032604c11a48./sbin/python./sbin/python.-m#ipykernel_launcher associated with connection .jvsc74a57bd0d2152fd7f0bbc62aa1baff8c990435d1e2c7175d001561303988032604c11a48./sbin/python./sbin/python.-m#ipykernel_launcher 
23:49:58.846 [debug] Disposing controller .jvsc74a57bd0d2152fd7f0bbc62aa1baff8c990435d1e2c7175d001561303988032604c11a48./sbin/python./sbin/python.-m#ipykernel_launcher (Interactive) associated with connection .jvsc74a57bd0d2152fd7f0bbc62aa1baff8c990435d1e2c7175d001561303988032604c11a48./sbin/python./sbin/python.-m#ipykernel_launcher 
23:49:58.847 [debug] Disposing controller .jvsc74a57bd0767d51c1340bd893661ea55ea3124f6de3c7a262a8b4abca0554b478b1e2ff90./usr/bin/python./usr/bin/python.-m#ipykernel_launcher associated with connection .jvsc74a57bd0767d51c1340bd893661ea55ea3124f6de3c7a262a8b4abca0554b478b1e2ff90./usr/bin/python./usr/bin/python.-m#ipykernel_launcher 
23:49:58.847 [debug] Disposing controller .jvsc74a57bd0767d51c1340bd893661ea55ea3124f6de3c7a262a8b4abca0554b478b1e2ff90./usr/bin/python./usr/bin/python.-m#ipykernel_launcher (Interactive) associated with connection .jvsc74a57bd0767d51c1340bd893661ea55ea3124f6de3c7a262a8b4abca0554b478b1e2ff90./usr/bin/python./usr/bin/python.-m#ipykernel_launcher 
23:49:58.847 [debug] Disposing controller .jvsc74a57bd04e1d9a8909477db77738c33245c29c7265277ef753467dede8cf3f814cde494e./usr/sbin/python./usr/sbin/python.-m#ipykernel_launcher associated with connection .jvsc74a57bd04e1d9a8909477db77738c33245c29c7265277ef753467dede8cf3f814cde494e./usr/sbin/python./usr/sbin/python.-m#ipykernel_launcher 
23:49:58.847 [debug] Disposing controller .jvsc74a57bd04e1d9a8909477db77738c33245c29c7265277ef753467dede8cf3f814cde494e./usr/sbin/python./usr/sbin/python.-m#ipykernel_launcher (Interactive) associated with connection .jvsc74a57bd04e1d9a8909477db77738c33245c29c7265277ef753467dede8cf3f814cde494e./usr/sbin/python./usr/sbin/python.-m#ipykernel_launcher 
23:49:58.848 [debug] Cleaning up server cache 4ff726d2-21df-4b04-bdd7-2e8fa1903e1e
23:49:58.848 [debug] Finished disposing HostJupyterExecution  4ff726d2-21df-4b04-bdd7-2e8fa1903e1e
